libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2420,8 +2433,26 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	// TODO some members might be pure, but for example `address(0x123).balance` is not pure
 	// although every subexpression is, so leaving this limited for now.
 	if (auto tt = dynamic_cast<TypeType const*>(exprType))
+	{
 		if (tt->actualType()->category() == Type::Category::Enum)
 			annotation.isPure = true;
+		else if (tt->actualType()->category() == Type::Category::Contract)
```

Comment:
This again does not catch `function() x = A.f`. I think that information has to be added to the type. We should discuss if we want that.

Reply:
Ah, sorry, that's wrong - there is a test about exactly that :)